packages/react-native/ReactCommon/react/performance/timeline/PerformanceEntryReporter.h

```diff
@@ -93,49 +33,64 @@ class PerformanceEntryReporter {
-  void startReporting(PerformanceEntryType entryType);
-  void stopReporting(PerformanceEntryType entryType);
-  void stopReporting();
-  void setAlwaysLogged(PerformanceEntryType entryType, bool isAlwaysLogged);
-  void setDurationThreshold(
-      PerformanceEntryType entryType,
-      double durationThreshold);
-
-  PopPendingEntriesResult popPendingEntries();
+  [[nodiscard]] PerformanceObserverRegistry& getObserverRegistry() {
+    return *observerRegistry_;
+  }
 
-  void logEntry(const PerformanceEntry& entry);
+  /*
+   * DOM Performance (High Resolution Time)
+   * https://www.w3.org/TR/hr-time-3/#dom-performance
+   */
+  // https://www.w3.org/TR/hr-time-3/#now-method
+  [[nodiscard]] DOMHighResTimeStamp getCurrentTimeStamp() const;
 
-  PerformanceEntryBuffer& getBuffer(PerformanceEntryType entryType) {
-    return buffers_[static_cast<int>(entryType) - 1];
+  void setTimeStampProvider(std::function<double()> provider) {
+    timeStampProvider_ = std::move(provider);
   }
 
-  const PerformanceEntryBuffer& getBuffer(
-      PerformanceEntryType entryType) const {
-    return buffers_[static_cast<int>(entryType) - 1];
-  }
+  /*
+   * Performance Timeline functions
+   * https://www.w3.org/TR/performance-timeline
+   */
+  // https://www.w3.org/TR/performance-timeline/#queue-a-performanceentry
+  void pushEntry(const PerformanceEntry& entry);
+
+  // https://www.w3.org/TR/performance-timeline/#getentries-method
+  // https://www.w3.org/TR/performance-timeline/#getentriesbytype-method
+  // https://www.w3.org/TR/performance-timeline/#getentriesbyname-method
+  [[nodiscard]] std::vector<PerformanceEntry> getEntries(
+      std::optional<PerformanceEntryType> entryType = std::nullopt,
+      std::string_view entryName = {}) const;
 
-  bool isReporting(PerformanceEntryType entryType) const {
-    return getBuffer(entryType).isReporting;
-  }
+  void logEventEntry(
+      std::string name,
+      double startTime,
+      double duration,
+      double processingStart,
+      double processingEnd,
+      uint32_t interactionId);
 
-  bool isAlwaysLogged(PerformanceEntryType entryType) const {
-    return getBuffer(entryType).isAlwaysLogged;
-  }
+  void logLongTaskEntry(double startTime, double duration);
 
-  uint32_t getDroppedEntriesCount() const {
-    return droppedEntriesCount_;
+  /*
+   * Event Timing API functions
+   * https://www.w3.org/TR/event-timing/
+   */
+  // https://www.w3.org/TR/event-timing/#dom-performance-eventcounts
+  [[nodiscard]] const std::unordered_map<std::string, uint32_t>& getEventCounts() const {
+    return eventCounts_;
   }
 
+  /*
+   * User Timing Level 3 functions
+   * https://w3c.github.io/user-timing/
+   */
+  // https://w3c.github.io/user-timing/#mark-method
   void mark(
       const std::string& name,
-      const std::optional<double>& startTime = std::nullopt);
+      const std::optional<DOMHighResTimeStamp>& startTime = std::nullopt);
```

Comment:
Nit: we could just defined this like this:
```suggestion
      DOMHighResTimeStamp startTime = getCurrentTimeStamp());
```